src/Angor/Client/Pages/Signatures.razor

```diff
@@ -357,26 +440,101 @@
                 sigJson);
 
             FounderProject.LastRequestForSignaturesTime = SignService.SendSignaturesToInvestor(encryptedContent, nostrPrivateKeyHex, signature.investorPubKey, signature.EventId);
-
             Storage.UpdateFounderProject(FounderProject);
 
             signaturesRequests.Single(_ => _.investorPubKey == signature.investorPubKey && _.TimeApproved is null)
                 .TimeApproved = FounderProject.LastRequestForSignaturesTime;
 
             return new OperationResult { Success = true };
-        });
+        }
+        catch (Exception ex)
+        {
+            return new OperationResult { Success = false, Message = $"An error occurred: {ex.Message}" };
+        }
+    }
+
+
+
 
-        if (operationResult.Success)
+
+
+    private async Task ApproveSignature(SignatureRequest signature)
+    {
+        var words = await passwordComponent.GetWalletAsync();
+        if (words == null)
+        {
+            passwordComponent.ShowPassword(async () =>
+            {
+                await ApproveSignature(signature);
+            });
+            return;
+        }
+
+        var operationResult = await PerformSignatureApproval(signature, words);
+
+        if (!operationResult.Success)
+        {
+            notificationComponent.ShowErrorMessage(operationResult.Message);
+        }
+        else
+        {
+            StateHasChanged();
+        }
+    }
+
+
+
+
+
+    private async Task ApproveAllSignatures()
```

Comment:
I think we have too many nested methods that we don't need, here is the flow I would suggest.

 `ApproveAllSignatures` prompt password if it is not there, then `ProcessSignatures` fetches words (so if password expires you don't bail out of the meth half way.
`ApproveSignatureCheckPassword` can go away because you already check and prompt in `ApproveAllSignatures`.
`ApproveSignature` takes the words as parameter and calls `PerformSignatureApproval` maybe even that method is a step too much) and that will show error message if sig fails.